feat: action outputs

AuHau opened the pull request:
Closes #351

gr2m commented on README.md:
```diff
@@ -90,6 +90,11 @@ If there are changes, it does the following
+
+ - `pr-number` - number of created/updated PR
+ - `result` - `created` or `updated` depending the PR was created or only updated
+
```
another result could also be "no change":
https://github.com/gr2m/create-or-update-pull-request-action/blob/65a6b01b19b69d7865fcf484340b06548fc02e64/index.js#L81-L88

Should we add `unchanged` to the list of possible results? And in case of `unchanged`, should we set `pr-number` if a pull request exists, but has not been updated?

Also could you rename `pr-number` to `pull-request-number`? I try to avoid abbreviations, `pr` might not be obvious for people new to GitHub.

gr2m reviewed (commented):
Could you update [`.github/workflows/test.yml`](https://github.com/gr2m/create-or-update-pull-request-action/blob/65a6b01b19b69d7865fcf484340b06548fc02e64/.github/workflows/test.yml) to include a job testing the outputs? You could also update the existing jobs instead. You can add a step with an `if` condition to fail the job if the output is not what it should be

AuHau commented:
I have implemented your feedback. Please allow the tests to run to see if it works 😉  😅

I have added the `unchanged` result, but it does not return the PR number. I would leave it like this until somebody would need it. It would require quite a bit of some refactoring of the code to get the PR number already at that stage of the Action.

gr2m commented:
> I have added the `unchanged` result, but it does not return the PR number. I would leave it like this until somebody would need it. It would require quite a bit of some refactoring of the code to get the PR number already at that stage of the Action.

good call 👍🏼

gr2m commented on .github/workflows/test.yml:
```diff
@@ -22,11 +22,14 @@ jobs:
           ACTIONS_STEP_DEBUG: true
         with:
           commit-message: "Just testing [skip ci]"
+      - if: ${{ steps.run.outpus.result != 'updated' || steps.run.outpus.pull-request-number > 0 }}
```
typo

```suggestion
      - if: ${{ steps.run.outputs.result != 'updated' || steps.run.outputs.pull-request-number > 0 }}
```

gr2m commented on .github/workflows/test.yml:
```diff
@@ -53,6 +57,8 @@ jobs:
       - run: "git push https://x-access-token:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git :test-create-new-pull-request"
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - if: ${{ steps.run.outpus.result != 'created' || steps.run.outpus.pull-request-number > 0 }}
+        run: 'echo "Output of action is not as expected" && exit 1'
```
I think the right condition here would be

```suggestion
      - if: ${{ !(steps.run.outputs.result == 'created' && steps.run.outputs.pull-request-number > 0) }}
        run: 'echo "Output of action is not as expected" && exit 1'
```

Or split it up into two steps to make it easier to read

```suggestion
      - if: ${{ steps.run.outputs.result != 'created' }}
        run: 'echo "result output is  \"${{ steps.run.outputs.result }}\" but expected \"created\"" && exit 1'
      - if: ${{ steps.run.outputs.pull-request-number > 0 }}
        run: 'echo "pull-request-number output is  \"${{ steps.run.outputs.pull-request-number }}\" but expected a number" && exit 1'
```
